internal/cache/rpc_cache.go

```diff
@@ -0,0 +1,43 @@
+package cache
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-redis/cache/v9"
+	"github.com/redis/go-redis/v9"
+	"github.com/samber/lo"
+	"github.com/satsuma-data/node-gateway/internal/jsonrpc"
+)
+
+var methodsToCache = []string{"eth_getTransactionReceipt"}
+
+const DefaultTTL = 30 * time.Minute
+
+func NewRPCCache(url string) *RPCCache {
+	if url == "" {
+		return nil
+	}
+
+	rdb := redis.NewClient(&redis.Options{
+		Addr: url,
+	})
+
+	return &RPCCache{
+		cache.New(&cache.Options{
+			Redis: rdb,
+		}),
+	}
+}
+
+type RPCCache struct {
+	*cache.Cache
+}
+
+func (c *RPCCache) ShouldCacheMethod(method string) bool {
+	return lo.Contains(methodsToCache, method)
+}
+
+func (c *RPCCache) GetKeyFromRequestBody(requestBody jsonrpc.SingleRequestBody) string {
+	return fmt.Sprintf("%s:%v", requestBody.Method, requestBody.Params)
```

Comment:
This is different than in the RFC.  Originally wanted to hash the method + parameter and call it a day.

However, I realized while testing that it became very difficult to find the cached result via the hashed cache key.

Redis has an idea of "[namespacing](https://redis.com/blog/5-key-takeaways-for-developing-with-redis/)" keys that fits our usecase well.

There are definitely some more things to figure out down the line when we get methods that have more parameters, but this works great for transaction receipts right now.

For example, the cache keys look like: `eth_getTransactionReceipt:[0x3a6f67beb73d07b1dd10c12de79767b6009f7b351ba1fe6282040aa6c57afef1]`

Reply:
What was hard about finding the cached result using the hashed cache key?

I've used that type of namespacing before and it's pretty idiomatic; sgtm!

Also, is this implicitly relying on `requestBody.Params` stringification? Seems like it works just fine for an array with 1 value. It's not impt now, but curious what it looks like for more complicated params data structures. Either way, consider adding an implicit toString on `requestBody.Params` so it's obvious it's casting an object into a string.

Reply:
Main scenario was trying to find the cached value for a request I just sent.  I would have to calculate the hash using the method + parameter and then run `get <key>` using redis-cli(especially needed with a lot of keys already set).

Yeah, I'm relying on Golang to do the string conversion for me here.  One thing to note is that I'm using the `%v` format verb for `Params`, which will give the "default form"(from reading docs this already does string conversion on the elements of the array).  This is nice to use because `Params` is `[]any`.

I definitely would need to test this more before I use it with more complex parameters though(and add sorting in the future).

Reply:
> I would have to calculate the hash using the method + parameter and then run get <key> using redis-cli(especially needed with a lot of keys already set).

Is the extra step compared to current state is hashing to get the key? Sorry, I'm not following what's tricky about that. (This approach lgtm, just wanted to made sure I understood everything you said.)

Reply:
> Is the extra step compared to current state is hashing to get the key? 

Yeah, would need to manually generate the hash before looking in the cache.  Having the human readable cache key is quite nice as long as it does not affect perf and resource usage.

I read up on some articles about how key length affects memory usage in Redis and I think getTransactionReceipt and getBlockByXXXX will still be fine using this approach.  getLogs will need some more brainstorming though.